pkg/webhook/statefulset/validating/statefulset_validation.go

```diff
@@ -358,3 +363,48 @@ func validateTemplateInPlaceOnly(oldTemp, newTemp *v1.PodTemplateSpec) error {
 
 	return nil
 }
+
+// ValidateVolumeClaimTemplateUpdate tests if only size expand when sc allow expansion.
+func ValidateVolumeClaimTemplateUpdate(c client.Client, sts, oldSts *appsv1beta1.StatefulSet) field.ErrorList {
+	if sts.Spec.VolumeClaimUpdateStrategy.Type == "" ||
+		sts.Spec.VolumeClaimUpdateStrategy.Type == appsv1beta1.OnPVCDeleteVolumeClaimUpdateStrategyType {
+		return nil
+	}
+	if len(sts.Spec.VolumeClaimTemplates) != len(oldSts.Spec.VolumeClaimTemplates) {
+		return field.ErrorList{field.Invalid(field.NewPath("spec", "volumeClaimTemplates"), sts.Spec.VolumeClaimTemplates, "volumeClaimTemplate can not be added or deleted when OnRollingUpdate")}
+	}
+	for i := range sts.Spec.VolumeClaimTemplates {
+		idxStr := fmt.Sprintf("volumeClaimTemplates[%v]", i)
```

Comment:
is it possible that pvc order is changed ?

Reply:
I will check it.